caddyhttp/proxy/reverseproxy.go

```diff
@@ -185,6 +186,8 @@ func NewSingleHostReverseProxy(target *url.URL, without string, keepalive int) *
 		rp.Transport = &http.Transport{
 			Dial: socketDial(target.String()),
 		}
+	} else if target.Scheme == "quic" {
+		rp.Transport = &h2quic.QuicRoundTripper{}
```

Comment:
What about all the stuff the director func above does? Does the QUIC transport not need/use it?